jobs/rocoto/anal.sh

```diff
@@ -4,9 +4,40 @@ source "${HOMEgfs}/ush/preamble.sh"
 
 ###############################################################
 # Source FV3GFS workflow modules
-. ${HOMEgfs}/ush/load_fv3gfs_modules.sh
-status=$?
-[[ ${status} -ne 0 ]] && exit ${status}
+#. ${HOMEgfs}/ush/load_fv3gfs_modules.sh
+#status=$?
+#[[ ${status} -ne 0 ]] && exit ${status}
+
+# TODO: clean this up
+#Source appropriate modulefiles based on machine
+
+source "${HOMEgfs}/ush/detect_machine.sh"
+if [[ "${MACHINE_ID}" = "wcoss2" ]]; then
+    # Source FV3GFS workflow modules
+    . "${HOMEgfs}"/ush/load_fv3gfs_modules.sh
+    status=$?
+    [[ ${status} -ne 0 ]] && exit "${status}"
+else
+    # Append compiler (only on machines that have multiple compilers)
+    COMPILER=${COMPILER:-"intel"}
+    if [[ "${MACHINE_ID}" = "hera" ]] || [[ "${MACHINE_ID}" = "cheyenne" ]]; then
+        MACHINE_ID=${MACHINE_ID}.${COMPILER}
+    fi
+
+    # Source machine specific GSI-EnKF modules
+    set +x
+    source "${HOMEgfs}/ush/module-setup.sh"
+    module use "${HOMEgfs}/sorc/gsi_enkf.fd/modulefiles"
+    module load gsi_"${MACHINE_ID}"
+
+    if [[ "${MACHINE_ID}" = "orion" ]]; then
+       conda activate regional_workflow
+    fi
```

Comment:
Please try this and see if it works in CI:
```suggestion
    if [[ "${MACHINE_ID}" = "orion" ]]; then
       module load hpc-miniconda/4.12.0
       conda activate regional_workflow
    fi
```

Reply:
It is my understanding that developers are not permitted to trigger gw CI.   I did, however, make the suggested change and rewound and rebooted 2021080118 gdasanal from my Orion account.   The job failed with 
```
+ anal.sh[23]: [[ orion = \c\h\e\y\e\n\n\e ]]
+ anal.sh[28]: set +x
Resetting modules to system default. Reseting $MODULEPATH back to system default. All extra directories will be removed from $MODULEPATH.
sh: line 1: 273469 Aborted                 /bin/lua /apps/lmod/lmod/libexec/lmod bash --regexp --no_redirect --spider_timeout 2.0 spider '^hpc%-miniconda/4%.12%.0$' 2> /dev/null
sh: line 1: 273472 Aborted                 /bin/lua /apps/lmod/lmod/libexec/lmod bash --regexp --no_redirect --spider_timeout 2.0 spider '^hpc%-miniconda/4%.12%.0$' 2> /dev/null
Lmod has detected the following error: These module(s) or extension(s) exist
but cannot be loaded as requested: "hpc-miniconda/4.12.0"
   Try: "module spider hpc-miniconda/4.12.0" to see how to load the module(s).
```
From previous runs in which I did not attempt to activate a conda environment, the following modules are loaded by `gsi_orion.lua` and `gsi_common.lua`
```
Currently Loaded Modules:
  1) contrib/0.1         10) netcdf/4.7.4  19) wrf_io/1.2.0
  2) miniconda3/4.12.0   11) bufr/11.7.0   20) ncio/1.1.2
  3) hpc/1.2.0           12) bacio/2.4.1   21) crtm/2.4.0
  4) intel/2022.1.2      13) w3emc/2.9.2   22) ncdiag/1.1.1
  5) hpc-intel/2022.1.2  14) sp/2.3.3      23) gsi_common
  6) impi/2022.1.2       15) ip/3.3.3      24) prod_util/1.2.2
  7) hpc-impi/2022.1.2   16) sigio/2.3.2   25) gsi_orion
  8) cmake/3.22.1        17) sfcio/1.4.1
  9) hdf5/1.10.6         18) nemsio/2.5.4
```

Reply:
@RussTreadon-NOAA My apologies and thanks for the clarification on CI.  I had a typo in the line I added.  It should be

`module load hpc-miniconda3/4.12.0`

That said, I'm starting to think this won't work, either.  Feel free to give it a try anyway, but I will also run some tests.

Reply:
As you suspect, adding "3" did not yield a successful run.  I'm back to the same `unbound variable` error as before

```
/apps/intel-2022.1.2/intel-2022.1.2/intelpython/latest/etc/conda/deactivate.d/xgboost_deactivate.sh: line 16: OCL_ICD_FILENAMES_RESET: unbound variable
End anal.sh at 14:02:23 with error code 1 (time elapsed: 00:00:08)
```

Since CI-Orion originally died in `gdasanal` with `getncdimlen` saying that it could not find `gsi_utils`, what if we update `PYTHONPATH` to include the path to `gsi_utils`.   
```
     if [[ "${MACHINE_ID}" = "orion" ]]; then
-       conda activate regional_workflow
+      wxflowPATH="${HOMEgfs}/ush:${HOMEgfs}/ush/python:${HOMEgfs}/ush/python/wxflow/src"
+      PYTHONPATH="${PYTHONPATH:+${PYTHONPATH}:}${wxflowPATH}"
+      export PYTHONPATH
     fi
```
`gsi_utils.py` resides in `$HOMEgfs/ush`.

The above change to `anal.sh` works when I rocotoboot 2021080118 gdasanal from my Orion account.  I don't know if this change would work from the account which runs CI-Orion.